wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -886,17 +887,19 @@ private TypeSpec enumAdapter(NameAllocator nameAllocator, EnumType type, ClassNa
         .addStatement("return $T.UINT32.encodedSize(toValue($N))", ProtoAdapter.class, value)
         .build());
 
-    builder.addMethod(MethodSpec.methodBuilder("encode")
-        .addAnnotation(Override.class)
-        .addModifiers(PUBLIC)
-        .addParameter(ProtoWriter.class, writer)
-        .addParameter(javaType, value)
-        .addException(IOException.class)
-        .addStatement("int $N = toValue($N)", i, value)
-        .addStatement("if ($N == $L) throw new $T($S + $N)",
-            i, -1, ProtocolException.class, "Unexpected enum constant: ", value)
-        .addStatement("$N.writeVarint32($N)", writer, i)
-        .build());
+    for (boolean reverse : Arrays.asList(false, true)) {
```

Comment:
maybe we could locally define a method and call it twice in a row with a named parameter to make it more reasonable

Reply:
hmmmm Java™

Reply:
Fixed!